[BUG] NPE in ShardSplittingQuery for segments with no live docs

### Describe the bug

For segments with no live docs (all tombstone docs), there doesn't seem to be ID field so the terms return null causing a NullPointerException when trying to split a shard. This happens if a user has significant amount of deleted docs and somehow landed with a segment where all docs are tombstone docs

Ref https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/shard/ShardSplittingQuery.java#L116

Exception trace:
```
Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.lucene.index.Terms.iterator()" because "terms" is null
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.index.shard.ShardSplittingQuery.findSplitDocs(ShardSplittingQuery.java:224)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.index.shard.ShardSplittingQuery$1.scorer(ShardSplittingQuery.java:115)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.FrozenBufferedUpdates.applyQueryDeletes(FrozenBufferedUpdates.java:394)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.FrozenBufferedUpdates.apply(FrozenBufferedUpdates.java:171)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter.forceApply(IndexWriter.java:6287)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter.tryApply(IndexWriter.java:6187)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter.lambda$publishFrozenUpdates$12(IndexWriter.java:2818)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter$EventQueue.processEventsInternal(IndexWriter.java:338)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter$EventQueue.processEvents(IndexWriter.java:327)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter.processEvents(IndexWriter.java:6030)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter.prepareCommitInternal(IndexWriter.java:3698)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter.commitInternal(IndexWriter.java:4134)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.apache.lucene.index.IndexWriter.commit(IndexWriter.java:4096)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.index.shard.StoreRecovery.addIndices(StoreRecovery.java:260)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.index.shard.StoreRecovery.lambda$recoverFromLocalShards$4(StoreRecovery.java:183)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:--
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.core.action.ActionListener$6.onFailure(ActionListener.java:311)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.index.shard.StoreRecovery.lambda$recoveryListener$10(StoreRecovery.java:628)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.core.action.ActionListener$1.onFailure(ActionListener.java:90)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.core.action.ActionListener.completeWith(ActionListener.java:347)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.index.shard.StoreRecovery.recoverFromLocalShards(StoreRecovery.java:173)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.index.shard.IndexShard.recoverFromLocalShards(IndexShard.java:3074)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.index.shard.IndexShard.lambda$startRecovery$37(IndexShard.java:4057)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.action.ActionRunnable$2.doRun(ActionRunnable.java:89)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-207.gz:  at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52)
elasticsearch.log-1.2026-05-11-01.ip-10-212-68-20
```



### Related component

_No response_

### To Reproduce

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


### Expected behavior

Should not throw NullPointerException

### Additional Details

**Plugins**
Please list all plugins currently enabled.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - OS: [e.g. iOS]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/server/src/main/java/org/opensearch/index/shard/ShardSplittingQuery.java
+++ b/server/src/main/java/org/opensearch/index/shard/ShardSplittingQuery.java
@@ -223,8 +223,11 @@ final class ShardSplittingQuery extends Query {
 
     private static void findSplitDocs(String idField, Predicate<BytesRef> includeInShard, LeafReader leafReader, IntConsumer consumer)
         throws IOException {
-        Terms terms = leafReader.terms(idField);
-        TermsEnum iterator = terms.iterator();
+        final Terms terms = leafReader.terms(idField);
+        if (terms == null) {
+            return;
+        }
+        final TermsEnum iterator = terms.iterator();
         BytesRef idTerm;
         PostingsEnum postingsEnum = null;
         while ((idTerm = iterator.next()) != null) {

--- a/server/src/test/java/org/opensearch/index/shard/ShardSplittingQueryTests.java
+++ b/server/src/test/java/org/opensearch/index/shard/ShardSplittingQueryTests.java
@@ -32,6 +32,7 @@
 package org.opensearch.index.shard;
 
 import org.apache.lucene.document.Field;
+import org.apache.lucene.document.NumericDocValuesField;
 import org.apache.lucene.document.SortedNumericDocValuesField;
 import org.apache.lucene.document.StringField;
 import org.apache.lucene.index.DirectoryReader;
@@ -50,12 +51,14 @@ import org.apache.lucene.util.BytesRef;
 import org.opensearch.Version;
 import org.opensearch.cluster.metadata.IndexMetadata;
 import org.opensearch.cluster.routing.OperationRouting;
+import org.opensearch.common.lucene.Lucene;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.index.mapper.IdFieldMapper;
 import org.opensearch.index.mapper.NestedPathFieldMapper;
 import org.opensearch.index.mapper.RoutingFieldMapper;
 import org.opensearch.index.mapper.SeqNoFieldMapper;
 import org.opensearch.index.mapper.Uid;
+import org.opensearch.index.mapper.VersionFieldMapper;
 import org.opensearch.test.OpenSearchTestCase;
 
 import java.io.IOException;
@@ -64,6 +67,47 @@ import java.util.Arrays;
 import java.util.List;
 
 public class ShardSplittingQueryTests extends OpenSearchTestCase {
+
+    public void testSplitOnNoopOnlySegment() throws IOException {
+        SeqNoFieldMapper.SequenceIDFields sequenceIDFields = SeqNoFieldMapper.SequenceIDFields.emptySeqID();
+        sequenceIDFields.tombstoneField.setLongValue(1);
+        Directory dir = newFSDirectory(createTempDir());
+        RandomIndexWriter writer = new RandomIndexWriter(random(), dir);
+        writer.addDocument(
+            Arrays.asList(
+                sequenceIDFields.seqNo,
+                sequenceIDFields.seqNoDocValue,
+                sequenceIDFields.primaryTerm,
+                sequenceIDFields.tombstoneField,
+                new NumericDocValuesField(VersionFieldMapper.NAME, 1),
+                Lucene.newSoftDeletesField()
+            )
+        );
+        writer.commit();
+        writer.close();
+
+        IndexMetadata metadata = IndexMetadata.builder("test")
+            .settings(Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT))
+            .numberOfShards(2)
+            .setRoutingNumShards(2 * 1000000)
+            .numberOfReplicas(0)
+            .build();
+        try (IndexReader reader = DirectoryReader.open(dir)) {
+            IndexSearcher searcher = new IndexSearcher(reader);
+            searcher.setQueryCache(null);
+            final Weight splitWeight = searcher.createWeight(
+                searcher.rewrite(new ShardSplittingQuery(metadata, 0, false)),
+                ScoreMode.COMPLETE_NO_SCORES,
+                1f
+            );
+            for (LeafReaderContext ctx : reader.leaves()) {
+                assertNull(ctx.reader().terms(IdFieldMapper.NAME));
+                Scorer scorer = splitWeight.scorer(ctx);
+                assertEquals(DocIdSetIterator.NO_MORE_DOCS, scorer.iterator().nextDoc());
+            }
+        }
+        dir.close();
+    }
 
     public void testSplitOnID() throws IOException {
         SeqNoFieldMapper.SequenceIDFields sequenceIDFields = SeqNoFieldMapper.SequenceIDFields.emptySeqID();

